frontend/src/store/modules/project.js

```diff
@@ -151,6 +151,10 @@ export default {
         addNewSceneObject(state, sceneObject) {
             state.sceneObjects.push(sceneObject);
         },
+
+        filterSceneObjects(state, filterFunction) {
+            state.sceneObjects = state.sceneObjects.filter(filterFunction);
```

Comment:
Tak jak to opisane w poprzednim komentarzu tutaj nie powinniśmy usuwać żadnych obiektów tylko zmienne powiązane. Więc trzeba zmienić filter na forEach lub map